editor/scene_tree_editor.cpp

```diff
@@ -526,6 +526,10 @@ void SceneTreeEditor::_node_removed(Node *p_node) {
 }
 
 void SceneTreeEditor::_node_renamed(Node *p_node) {
+	if (!get_scene_node()->is_a_parent_of(p_node)) {
```

Comment:
> Yes for 3.3-stable it's too risky, but it can likely make its way into 3.4 and 3.3.1.

Very true, I think I still somehow managed to get one of these three lines wrong :stuck_out_tongue: 

---

Of course this is a very simple change but my main concern is whether to use `tree` here or `get_scene_node()`. If I'm correct, `tree` is the representation of the nodes and `get_scene_node()` contains the nodes themselves.

Reply:
This would need a rebase and change to `is_ancestor_of()` (renamed).

I'm also unsure on whether that's the right API to do this bugfix, but it does seem sensible. I think we can merge and see if there's any regression :)